x-pack/plugins/security_solution/common/endpoint/types/index.ts

```diff
@@ -1120,6 +1120,10 @@ export interface BlocklistFields {
   blocklist: boolean;
 }
 
+export interface OnWriteScanFields {
+  on_write_scan: boolean;
```

Comment:
For serverless: even though we are backfilling this field correctly, it will be `undefined` for a window of time because the UI goes live before the migrations are run.

So it's possible the UI gets `on_write_scan: undefined` maps to false-y unless we provide a default for it somewhere in our application code which I don't believe we are doing.

We should set `on_write_scan?: boolean` then after this version is released we can change it to `on_write_scan: boolean` as a two stage release.

This is currently missing from our developer documentation, even though it's thoroughly covered in our RFCs etc for the new migration system.

Reply:
thanks @jloleysens for the input!

changed the parameter to optional, and added default handling here: 717341d1496dc04370c62ba92269eafa083cbfa4
it is only handled on the client side, because I think that is what matters:
- when the user wants to edit the Endpoint Defend integration config, they will see `on_write_scan` enabled (as it is the default value)
- when fetching the package policy config via API, the field will be simply missing, no falsy value
- and most importantly: Endpoint integration on Elastic Agent will supposedly handle the missing `on_write_scan` field as `true`

also modified the migration script, so if the user edits the config before migration is performed, the backfill won't overwrite the added fields with the default value: ff86b4fbb9e1243db2fc0bea8eb545ac1ede4bf3

let me know if I'm missing something